threescale_api/defaults.py

```diff
@@ -386,6 +386,53 @@ def _invalidate(self):
         self._entity = None
 
 
+class DefaultPaginationClient(DefaultClient):
+    """ Client to handle API endpoints with pagination.
+    List of endpoints supporting pagination with per_page size:
+        - accounts 500
+        limits per app plan 50 - not implemented in client
+        application list for all services 500 - not implemented in client
+        - backend mapping rules 500
+        - backend method list 500
+        - backend metric 500
+        - backend 500
+        - service 500
+        invoice list by account 20 - not implemented by standard "list" method
+        - invoice list 20
+        - all cms 100
+    """
+    def __init__(self, *args, per_page=500, **kwargs):
+        self.per_page = per_page
+        super().__init__(*args, **kwargs)
+
+    def _list(self, **kwargs):
+        """ List all objects via paginated API endpoint """
+        if "page" in kwargs.get("params", {}) or self.per_page is None:
+            return super()._list(**kwargs)
+        pagenum = 1
+
+        kwargs = kwargs.copy()
+        if "params" not in kwargs:
+            kwargs["params"] = {}
```

Comment:
You can define params using setdefault before this if and then reuse it afterwards